### PR TITLE
Yeheng_Sun_S4548085_COMP3710_ADNIClassifier

### PR DESCRIPTION
**Pull request by SteveSun1015**
The code is written to provide a solution for problem 6 which is Classify Alzheimer’s disease (normal and AD) of the ADNI brain dataset using a visual transformer.

Commit log explanation:
The "Prepare Data" download the Alzheimer’s disease images from "https://cloudstor.aarnet.edu.au/plus/s/L6bbssKhUoUdTSI/download".

The "finished dataset.py" loads all the images by opencv

The "init need py file, draft module.py" constructs the structure of the root directory with the need. py file. Also codes the structure of the model.

The "draft train.py and start training" finished the train.py script and started training the model.

The "Use dataloader to load data" change the code in dataset.py as used opencv to load all the images before training need a lot of ram which makes the training cannot have a good image size and batch size. So I use a tensorflow image loader to load the images instead of an opencv.

The "Final Draft" saves the model. h5 (weights of the model), draft the predict.py and readme

The "Finish predict.py and visualisation" and "finish Read me" finished the predict.py which load the weight and evaluate the model and finished the readme.md

The "Delete image dataset from git" deletes all the images for training and testing.

---------------------------------------------------------------------------------------------------------------4

FINAL CHECK
As I run the script in a cloud server, I found that there are some mistakes in hyperparameters and modules compared to the script in cloud. So some new commits change the hyperparameters and module

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
This is an initial inspection, no action is required at this point

ViT design: Great
Preprocessing: Only did train, validation split. Need a test set after training
Model training: Great
Commit log: Great

**Comment by SteveSun1015** [comment]
> This is an initial inspection, no action is required at this point
> 
> ViT design: Great Preprocessing: Only did train, validation split. Need a test set after training Model training: Great Commit log: Great

Hi Linfeng, I think I probably do have the test set by the code "test_ds = dataset.createTestData()" at the predict.py file, could you please double check it for me? Thanks

**Comment by LinfengLiu98** [comment]
> > This is an initial inspection, no action is required at this point
> > ViT design: Great Preprocessing: Only did train, validation split. Need a test set after training Model training: Great Commit log: Great
> 
> Hi Linfeng, I think I probably do have the test set by the code "test_ds = dataset.createTestData()" at the predict.py file, could you please double check it for me? Thanks

Do you have a validation set? So when you do the data split, you need to create train, validation and test set. If you have, please clarify in the README: e.g.: Training set: xxxxx number of images, Validation set: xxxxx number of images, Test set: xxxxx number of images. From what I can see you have a training and a val/test set. If you use the test set during training that's actually the validation set then you are missing a test set no matter how you call it. If you use the test set after training then you are missing a validation set.

**Comment by SteveSun1015** [comment]
Hi Linfeng,&nbsp;
Yes, I only have train set and test set. But when I define training data set I define 30% of the training set as validation set and I also state this in readme file.&nbsp;
Below are two screenshots from readme and datase.py.
a0dd4604839bc9f0d4034ea8b978a90
e26a60b314d723e5deb9a35f884cbff

**Comment by shakes76** [comment]
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate use and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem (no testing results) -2
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage 
Module present
Commenting
No Data leakage (no test set) -1
Difficulty: Hard
## Commit Log
Meaningful commit messages, could be more informative -1
Progressive commits used
## Documentation
ReadMe acceptable, no architecture -1
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, remove model weights -2
Request Description good
Late -12 (restored if extension was approved)
